plugins/network-elements/juniper-srx/src/com/cloud/network/resource/JuniperSrxResource.java

```diff
@@ -3611,7 +3611,7 @@ private String setDelete(String xml, boolean delete) {
 
     private Long getVlanTag(String vlan) throws ExecutionException {
         Long publicVlanTag = null;
-        if (!vlan.equals("untagged")) {
+        if (!vlan.equals("vlan://untagged")) {
```

Comment:
In APIs, untagged is a valid parameter value that is passed often. Can you consider changing this to `vlan.contains("untagged")`?